runtime/onert/api/python/package/experimental/train/optimizer/registry.py

```diff
@@ -0,0 +1,43 @@
+from .adam import Adam
+from .sgd import SGD
+
+
+class OptimizerRegistry:
+    """
+    Registry for creating optimizers by name.
+    """
+    _optimizers = {"adam": Adam, "sgd": SGD}
+
+    @staticmethod
+    def create_optimizer(name):
+        """
+        Create an optimizer instance by name.
+        Args:
+            name (str): Name of the optimizer.
+        Returns:
+            BaseOptimizer: Optimizer instance.
+        """
+        if name not in OptimizerRegistry._optimizers:
+            raise ValueError(
+                f"Unknown Optimizer: {name}. Custom optimizer is not supported yet")
+        return OptimizerRegistry._optimizers[name]()
+
+    @staticmethod
+    def map_optimizer_to_enum(optimizer_instance):
+        """
+        Maps an optimizer instance to the appropriate enum value.
+        Args:
+            optimizer_instance (Optimizer): An instance of an optimizer.
+        Returns:
+            optimizer_type: Corresponding enum value for the optimizer.
+        Raises:
+            TypeError: If the optimizer_instance is not a recognized optimizer type.
+        """
+        # Optimizer to Enum mapping
+        optimizer_to_enum = {SGD: "SGD", Adam: "ADAM"}
```

Comment:
As I asked offline, It looks like this function returns `string`, instead of `enum`. I wonder it is intentional or not.

Reply:
Thanks for your pointing out. I fixed it.